pennylane/hf/particle_number.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the functions needed for computing the particle number observable.
+"""
+import autograd.numpy as anp
+import pennylane as qml
+from pennylane import numpy as np
+from pennylane.hf.hamiltonian import _generate_qubit_operator, _return_pauli, simplify
+
+
+def particle_number(orbitals):
+    r"""Computes the particle number observable :math:`\hat{N}=\sum_\alpha \hat{n}_\alpha`
+    in the Pauli basis.
+
+    The particle number operator is given by
+
+    .. math::
+
+        \hat{N} = \sum_\alpha \hat{c}_\alpha^\dagger \hat{c}_\alpha,
+
+    where the index :math:`\alpha` runs over the basis of single-particle states
+    :math:`\vert \alpha \rangle`, and the operators :math:`\hat{c}^\dagger` and :math:`\hat{c}` are
+    the particle creation and annihilation operators, respectively.
+
+    Args:
+        orbitals (int): Number of *spin* orbitals. If an active space is defined, this is
+            the number of active spin-orbitals.
+
+    Returns:
+        pennylane.Hamiltonian: the particle number observable
+
+    **Example**
```

Comment:
I recently learned that adding Raises section to our doc-strings is a requirement by PL doc-string [standards](https://pennylane.readthedocs.io/en/stable/development/guide/documentation.html#functions-and-methods)!  

```suggestion
    Returns:
        pennylane.Hamiltonian: the particle number observable

    Raises:
        ValueError: If orbitals is less then or equal to 0

    **Example**
```

Reply:
Added.